perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_virt_elixirBonaFideStatus.java

```diff
@@ -0,0 +1,120 @@
+
+		try {
+			Attribute attributeDef = sess.getPerunBl().getAttributesManagerBl().getAttribute(sess, user, AttributesManager.NS_USER_ATTR_DEF + ":elixirBonaFideStatusREMS");
+			String attrDefValue = (String) attributeDef.getValue();
+			if (attrDefValue != null && !attrDefValue.isEmpty()) {
+				attribute.setValue(attrDefValue);
+			} else {
+				Attribute uesAttribute = sess.getPerunBl().getAttributesManagerBl().getAttribute(sess, user, AttributesManager.NS_USER_ATTR_VIRT + ":eduPersonScopedAffiliations");
+				List<String> affiliations = (List<String>)uesAttribute.getValue();
+				for (String affiliation : affiliations) {
+					if (affiliation != null && affiliation.startsWith("faculty@")) {
+						attribute.setValue(url);
+						break;
+					}
+				}
+			}
+		} catch (WrongAttributeAssignmentException | AttributeNotExistsException e) {
+			log.error("Cannot read elixirBonaFideStatusREMS from user " + user.getId(), e);
+		}
+		return attribute;
+	}
+
+	@Override
+	public List<String> resolveVirtualAttributeValueChange(PerunSessionImpl perunSession, String message) throws InternalErrorException, WrongReferenceAttributeValueException, AttributeNotExistsException, WrongAttributeAssignmentException {
+		List<String> resolvingMessages = new ArrayList<>();
+		if (message == null) return resolvingMessages;
+
+		Matcher addUserExtSourceMatcher = addUserExtSource.matcher(message);
+		Matcher removeUserExtSourceMatcher = removeUserExtSource.matcher(message);
+		Matcher extSourceRemsMatcher = extSourceRems.matcher(message);
+
+		User user;
+		Attribute attrVirtRemsLogins;
+
+		if(extSourceRemsMatcher.find()) {
+			if (addUserExtSourceMatcher.find() || removeUserExtSourceMatcher.find()) {
+				user = getUserFromMessage(message);
+				if (user != null) {
+					attrVirtRemsLogins = perunSession.getPerunBl().getAttributesManagerBl().getAttribute(perunSession, user, AttributesManager.NS_USER_ATTR_VIRT + ":elixirBonaFideStatus");
+					String messageAttributeSet = attrVirtRemsLogins.serializeToString() + " set for " + user.serializeToString() + ".";
+					resolvingMessages.add(messageAttributeSet);
+				}
+			}
+		}
+		return resolvingMessages;
+	}
+
+	@Override
+	public AttributeDefinition getAttributeDefinition() {
+		AttributeDefinition attr = new AttributeDefinition();
+		attr.setNamespace(AttributesManager.NS_USER_ATTR_VIRT);
+		attr.setFriendlyName("elixirBonaFideStatus");
+		attr.setDisplayName("Bona fide researcher status");
+		attr.setType(String.class.getName());
+		attr.setDescription("Flag if user is qualified researcher. URI ‘http://www.ga4gh.org/beacon/bonafide/ver1.0’ value is provided if person is bona fide researcher. Empty value otherwise.");
+		return attr;
+	}
+
+	/**
+	 * Get User from message if exists and there is only one. In other case return null instead.
+	 *
+	 * @param message
+	 * @return user or null
+	 * @throws InternalErrorException
+	 */
+	private User getUserFromMessage(String message) throws InternalErrorException {
```

Comment:
This method is now without duplicities (and with customizations) in ModulesUtilsBl, please use it instead.